mapstory/static/mapstory/js/search.js

```diff
@@ -221,7 +221,27 @@
             $scope.text_query = $location.search()['q'].replace(/\W+/g," ");
           }
           if ($location.search().hasOwnProperty('type__in')){
-            $scope.type__in = $location.search()['type__in'].replace(/\W+/g," ");;
+            $scope.type__in = $location.search()['type__in'].replace(/\W+/g," ");
+          }
+          // Filter by Regions
+          if ($location.search().hasOwnProperty('regions')) {
+            $scope.results = data.objects.filter(function(value) {
+              for (var index = 0; index < value.regions.length; index++) {
+                if ($location.search()['regions'].indexOf(value.regions[index]) > -1) {
+                  return true;
+                }
+              }
+            });
+          }
```

Comment:
Is this the correct way to be using the haystack filters? This fixes it to work again on the search page...